### PR TITLE
workflow version upgrade protoc & python

### LINKED ISSUE #2594: deprecated version of actions - python-client, actions.yml(python-tests) workflows
**Describe the bug**

`actions/checkout@v2`, `actions/setup-python@v2` and `arduino/setup-protoc@v1` actions have been deprecated.

**Files Or Markdown that can reproduce the issue**

image

┆Issue is synchronized with this [Jira Task](https://gr-oss.atlassian.net/browse/BATCH-383) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by nitishchauhan0022**
Fixes #2594 

#### Special notes for your reviewer:

┆Issue is synchronized with this [Jira Task](https://gr-oss.atlassian.net/browse/BATCH-534) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by nitishchauhan0022** [comment]
hey @kannon92, the version scheme of protobuf have changes last year. The version 21.0 is not the latest, should we go for the latest?
PTAL: https://protobuf.dev/news/2022-05-06/

 Also the version in the [go.mod  ](https://github.com/armadaproject/armada/blob/39d8f10c36638943b3f2f66aa4f0f433c580ed75/go.mod#L26)is deprecated 
 
image

**Comment by kannon92** [comment]
> hey @kannon92, the version scheme of protobuf have changes last year. The version 21.0 is not the latest, should we go for the latest? PTAL: https://protobuf.dev/news/2022-05-06/
> 
> Also the version in the [go.mod ](https://github.com/armadaproject/armada/blob/39d8f10c36638943b3f2f66aa4f0f433c580ed75/go.mod#L26)is deprecated
> 
> image

Please create a new ticket for that.  I'm fine with including it in build but I'd like to do the version bump separately.

File: .github/workflows/python-client.yml [hunk 1]
```diff
@@ -83,8 +83,8 @@ jobs:
           mkdir -p .kube/external
           go run github.com/magefile/mage@v1.14.0 -v localdev minimal
       - name: Install Protoc
-        uses: arduino/setup-protoc@v1
+        uses: arduino/setup-protoc@v2
         with:
-          version: '3.17.3'
+          version: '21.0'
```
Comment (richscott) [inline .github/workflows/python-client.yml:88]: Is this current? I believe we're using protoc 23.3.  Version 24.0 is the very latest, but it only just came out a couple days ago, so we may not want that.

Comment (nitishchauhan0022) [reply .github/workflows/python-client.yml]: Changed it to 23.3

**Review by kannon92 — APPROVED** [review]

**Pull request merged**